build/template-restore-build-libsandsamples.yaml

```diff
@@ -28,6 +28,14 @@ steps:
     custom: 'workload'
     arguments: 'restore .\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj'
 
+- task: Bash@3
+  displayName: Install MAUI
```

Comment:
This task adds ~6 minutes to the build time. Wondering if we can move these WAM devapps in to a separate sln, which we build and run these tests on?

It's ok for now to improve the tests. Just trying to be mindful each time we update the build pipeline to make sure we don't get into situation again when we have to refactor the pipeline since our builds take long time again.